chains/evm/calls/contracts/bridge/bridge.go

```diff
@@ -163,7 +163,7 @@ func (c *BridgeContract) Erc721Deposit(
 		Str("resourceID", hexutil.Encode(resourceID[:])).
 		Str("tokenID", tokenId.String()).
 		Msgf("ERC721 deposit")
-	data := deposit.ConstructErc721DepositData(recipient.Bytes(), tokenId, []byte(metadata))
+	data := deposit.ConstructErc721DepositData(recipient.Bytes(), tokenId, []byte(metadata), opts.Priority)
```

Comment:
I do not like that if Priority is not set it anyways being added to the calldata. 
I am thinking about adding if/esle to check the priority if it is == 0 then sue COnstructDepositData else COnstructDepositDataWithPriority 
@mpetrun5 @MakMuftic

Reply:
Or you can use https://www.sohamkamani.com/golang/options-pattern/  and implement WithPriority opts